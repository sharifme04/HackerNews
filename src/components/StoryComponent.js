import React, { useState,  useCallback} from 'react';
import PropTypes from 'prop-types';

const StoryComponent = ({stories}) =>{
  const [selected, setSelected] = useState(new Map());
  const onSelect = useCallback(
    id => {
      const newSelected = new Map(selected);
      newSelected.set(id, !selected.get(id));
      setSelected(newSelected);
    }, [selected]);

  return (
    <div className="gridContainer">
      {stories && stories.map(e =>
        <div
          key={e.id}
          className={`card mx-1 my-1 ${!!selected.get(e.id) ? "card-expand" :"card-actual"}`}
          onClick={()=>onSelect(e.id)}
        >
        <div className="row no-gutters mx-1 my-1">
        <div className="col-md-3 mt-3">
           <span style={{fontSize:"20px"}}>&#9650;</span>
         <p className="card-text">Score</p>
        </div>
        <div className="col-md-9">
         <div className="card-body">
           <h6 className="card-title">{e.title}</h6>
           <p className="card-text">By: {e.by}</p>
           <p className="card-text"><small>{e.text}</small></p>
         </div>
         </div>
        </div>
      </div>)}
    </div>
)};

StoryComponent.propTypes = {
  stories: PropTypes.array
}

export default StoryComponent;
