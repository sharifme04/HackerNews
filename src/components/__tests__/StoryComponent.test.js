import React from "react";
import { shallow } from "enzyme";
import StoryComponent from "../StoryComponent";

const mockstory = [
  {
    by: "jger15",
    descendants: 15,
    id: 22000126,
    kids: [
      22001188,
      22001236,
      22001202,
      22001107,
      22001179,
      22001091,
      22001154,
      22001106
    ],
    score: 38,
    time: 1578567242,
    title: "Work on these things",
    type: "story",
    url:
      "https://marginalrevolution.com/marginalrevolution/2019/12/work-on-these-things.html"
  },
  {
    by: "awwstn",
    descendants: 30,
    id: 22012831,
    kids: [
      22013151,
      22013052,
      22013186,
      22013149,
      22013020,
      22013011,
      22013299,
      22013028,
      22013270,
      22013182
    ],
    score: 51,
    time: 1578677960,
    title: "Postman is now charging 50% more for ~5x less",
    type: "story",
    url: "https://capiche.com/q/what-has-changed-in-postmans-new-pricing"
  }
];

describe("<StoryComponent />", () => {
  test("renders", () => {
    const component = shallow(<StoryComponent stories={mockstory} />);
    expect(component).toMatchSnapshot();
  });

  test("return the default array when there is no data to map through", () => {
    const component = shallow(<StoryComponent />);
    expect(component).toMatchSnapshot();
  });

  test("does not break without value", () => {
    const component = shallow(<StoryComponent />);
    expect(component.find("h6")).toHaveLength(0);
    expect(component.find("p")).toHaveLength(0);
    expect(component.find("small")).toHaveLength(0);
  });

  test("does not break with an empty array", () => {
    const component = shallow(<StoryComponent stories={[]} />);
    expect(component.find("h6")).toHaveLength(0);
    expect(component.find("p")).toHaveLength(0);
    expect(component.find("small")).toHaveLength(0);
  });
});
