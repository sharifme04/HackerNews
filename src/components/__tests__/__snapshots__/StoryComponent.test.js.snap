// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoryComponent /> renders 1`] = `
<div
  className="gridContainer columnNumber"
>
  <div
    className="card mx-1 my-1 card-actual"
    key="22000126"
    onClick={[Function]}
  >
    <div
      className="row no-gutters mx-1 my-1"
    >
      <div
        className="col-md-3 mt-3"
      >
        <span
          style={
            Object {
              "fontSize": "20px",
            }
          }
        >
          ▲
        </span>
        <p
          className="card-text"
        >
          Score
        </p>
      </div>
      <div
        className="col-md-9"
      >
        <div
          className="card-body"
        >
          <h6
            className="card-title"
          >
            Work on these things
          </h6>
          <p
            className="card-text"
          >
            By: 
            jger15
          </p>
          <p
            className="card-text"
          >
            <small>
              No text available
            </small>
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card mx-1 my-1 card-actual"
    key="22012831"
    onClick={[Function]}
  >
    <div
      className="row no-gutters mx-1 my-1"
    >
      <div
        className="col-md-3 mt-3"
      >
        <span
          style={
            Object {
              "fontSize": "20px",
            }
          }
        >
          ▲
        </span>
        <p
          className="card-text"
        >
          Score
        </p>
      </div>
      <div
        className="col-md-9"
      >
        <div
          className="card-body"
        >
          <h6
            className="card-title"
          >
            Postman is now charging 50% more for ~5x less
          </h6>
          <p
            className="card-text"
          >
            By: 
            awwstn
          </p>
          <p
            className="card-text"
          >
            <small>
              No text available
            </small>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<StoryComponent /> return the default array when there is no data to map through 1`] = `
<div
  className="gridContainer columnNumber"
/>
`;
